ts_src/manager.ts

```diff
@@ -125,6 +129,32 @@ export class GameManager {
         Settings.board_initialised = true;
     }
 
+    newNote(name: string, text: string, show: boolean, sync: boolean, uuid:string = ""): void {
```

Comment:
I have a slight preference to using 

```typescript
newNote(name: string, text: string, show: boolean, sync: boolean, uuid:string = ""): void {
    uuid = uuid || uuidv4();
```

Reply:
While that solution is more elegant is doesn't work in this case because of this line: `gameManager.notes.get(uuid).show();`

Typescript throws an error here because orderedMap.get expects a string as an argument and Typescript thinks that uuid can be a string or undefined.

Reply:
Ah yes because it's reassigned to uuid itself which is `string || undefined` my bad, the shape constructor assigns to this.uuid which is solely `string`.  I suppose you could use `new Note(name, text, uuid || uuidv4())`, but you may leave it as you have it now as well